File: 1014-best-sightseeing-pair/1014-best-sightseeing-pair.cpp
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& values) {
        int maxi = values[0];
        for(int i=1; i<values.size(); i++){
            maxi = max(maxi, values[0]+values[i]-i);
            values[0] = max(values[0], values[i]+i);
        }
        return maxi;
    }
};

File: 1014-best-sightseeing-pair/1014-best-sightseeing-pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        a = values[0]
        maxi = a
        for i in range(1, len(values)):
            maxi = max(maxi, a + values[i]-i)
            a = max(a, values[i]+i)
        return maxi
        